simuleval/evaluator/instance.py

```diff
@@ -235,10 +235,20 @@ def __init__(
         args: Optional[Namespace],
     ):
         super().__init__(index, dataloader, args)
+        self.args = args
         self.sample_rate_value = None
         self.sample_list = None
         self.source_finished_reading = False
         self.dataloader: SpeechToTextDataloader
+        if self.dataloader is not None:
+            self.tgt_lang = self.dataloader.tgt_lang
+            if isinstance(self.tgt_lang, list):
+                if index < len(self.tgt_lang):
+                    self.tgt_lang = self.tgt_lang[self.index]
+                else:
+                    self.tgt_lang = None
+        else:
+            self.tgt_lang = None
```

Comment:
This might work, but is not the cleanest way to implement the tgt_lang assignment for instances. 

Like I had pointed out earlier, we want something similar to how the source and target getters work.
https://github.com/facebookresearch/SimulEval/blob/main/simuleval/evaluator/instance.py#L46-L48
https://github.com/facebookresearch/SimulEval/blob/main/simuleval/data/dataloader/dataloader.py#L55-L56

In case it is not clear, we want to be able to do something like `self.tgt_lang = self.dataloader[index]["tgt_lang"]` to access the target language of a particular index.

Reply:
Another cause for concern here is that all dataloaders are going to have the `tgt_lang` attribute. 
There are some dataloaders (in internal fairseq repo), which don't have the `tgt_lang` attribute that you added in `SpeechToTextDataloader` or  `SpeechToSpeechDataloader`in this PR and  `self.dataloader.tgt_lang` would throw an exception if the dataloader doesn't have this attribute defined correctly. So, please keep in mind while making these changes that not all dataloader have the tgt_lang attribute and handle it accordingly.

Also, at this point feel free to factor out and move necessary changes to the base class `GenericDataloader` whenever it makes sense to do so.